### PR TITLE
fix shuffling for ShufflerIterDataPipe instances

### PR DESCRIPTION
**Pull request by loubnabnl**
fixes https://github.com/huggingface/accelerate/issues/375

cc @sgugger

### CODE REVIEW

File: src/accelerate/data_loader.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 import torch
 from torch.utils.data import BatchSampler, DataLoader, IterableDataset
+from torch.utils.data.datapipes.iter.combinatorics import ShufflerIterDataPipe
```
Comment (sgugger) [inline src/accelerate/data_loader.py:20]: This will break compatibility with older versions of PyTorch as I believe datapipes were just introduced in PyTorch 1.11.

File: src/accelerate/data_loader.py [hunk 2]
```diff
@@ -326,12 +327,15 @@ class DataLoaderDispatcher(DataLoader):
     """
 
     def __init__(self, dataset, split_batches: bool = False, **kwargs):
+        shuffle = dataset._shuffle_enabled if isinstance(dataset, ShufflerIterDataPipe) else False
```
Comment (sgugger) [inline src/accelerate/data_loader.py:330]: There should be a versioning test of PyTorch first, then this instance check. :-)

**Review by sgugger — COMMENTED** [review]
Looks good in principle! We just have to make a tiny bit more effort to keep supporting older PyTorch versions :-)

**Review by sgugger — APPROVED** [review]
Perfect, thanks a lot!

File: src/accelerate/data_loader.py [hunk 3]
```diff
@@ -326,12 +326,21 @@ class DataLoaderDispatcher(DataLoader):
     """
 
     def __init__(self, dataset, split_batches: bool = False, **kwargs):
+        shuffle = False
+        if version.parse(torch.__version__) >= version.parse("1.11.0"):
```
Comment (sgugger) [inline src/accelerate/data_loader.py:330]:
```suggestion
        if version.parse(torch.__version__) >= version.parse("1.11.99"):
```
Just in case we get many micro releases ;-)

Comment (sgugger) [reply src/accelerate/data_loader.py:330]: Arf, review before coffee. I read > and thought for a minute it needed PyTorch 1.12, you can ignore :-)
